Resources/Prototypes/Entities/Effects/puddle.yml

```diff
@@ -95,19 +102,24 @@
   parent: PuddleBase
   components:
   - type: Sprite
-    sprite: Fluids/splatter.rsi # Placeholder
-    state: splatter-0
     netsync: false
+    drawdepth: FloorObjects
+    layers:
+    - sprite: Fluids/splatter.rsi # Placeholder
+      map: ["enum.PuddleVisualLayers.Puddle"]
+      state: splatter-0
+    - sprite: Fluids/wet_floor_sparkles.rsi
+      map: ["enum.PuddleVisualLayers.WetFloorEffect"]
+      state: sparkles
+      visible: false
   - type: Puddle
     slipThreshold: 20
     overflowVolume: 50
     opacityModifier: 8
   - type: Evaporation
     evaporateTime: 400 # very slow
   - type: Appearance
-    visuals:
-    - type: PuddleVisualizer
-      recolor: true
+  - type: PuddleVisuals
 
 - type: entity
   name: vomit
```

Comment:
Vomit and toxic vomit shouldn't auto evaporate. so they either need something like
```yaml
  - type: Evaporation
    upperLimit: 5.01 # above minimum reachable by mopping, but smaller than starting volume
```
or they just need to initially not have an evaporation component (or at least a disabled one) , and get given one as a result of mopping. The second option seems like a better fix, especially for something like egg or tomato stains, which have a very small solution quantity.

Though maybe that's more mopping related than puddle visuals stuff. I haven't actually looked at #6718 yet, so if you already fix this there, then ignore this comment

Reply:
Actually based on the PR description it sounds like the plan for tomato stains is for it to never even leave behind evaporating sparkles, and to simply disappear.

In that case, the tomato puddles just shouldn't have an evaporation component in the first place, which would probably require either removing it from base puddle, or making tomatos not inherit from base puddle.

Reply:
For sure! I've got some YAML cleanup to do here anyway, so I think I'll make vomit, tomatoes, and any other "no-evaporation" puddles inherit from a different base.

Reply:
I also like what you've said about mopping giving the puddle an evaporationComponent if it doesn't have one.

Otherwise, if I simply make vomit/tomato puddles have no evaporationComponent, and a bunch of water gets added to the puddle, then it could be made to slip people without evaporating (which we want to avoid). I'll think on it some more as I'm going through all the changes.

Reply:
I've decided I want to tackle the stain/evaporation fixes in its own PR when and if this PR and my mopping ECS one are merged. It's best done holistically with changes to mopping, evaporation, and puddle behaviour.

For now I've brought vomit/tomato in line with what Blood puddles do, which is make them evaporate extremely slowly (1 unit every 400 seconds). It's a stopgap for now, but I'm focusing this PR on getting the visuals back to working order.